packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -754,7 +762,8 @@ function commitClassLayoutLifecycles(
     if (
       enableProfilerTimer &&
       enableProfilerCommitHooks &&
-      finishedWork.mode & ProfileMode
+      finishedWork.mode & ProfileMode &&
+      getExecutionContext() & CommitContext
```

Comment:
Nit: Since these checks repeat in a bunch of places, might make sense to extract it into a separate function at this point. So we don't forget this logic later. Since it'll be a small function it will get inlined by Closure anyway.